src/components/Button.tsx

```diff
@@ -0,0 +1,38 @@
+'use client';
+import { cva, type VariantProps } from 'class-variance-authority';
+import { cn } from '@/lib/cn';
+
+const buttonVariants = cva('cursor-pointer transition-all duration-200 flex items-center justify-center', {
+  variants: {
+    variant: {
+      default: 'rounded-md bg-green-100 font-bold text-lg text-white hover:bg-black-200',
+      outline:
+        'rounded-md border border-black-200 font-bold text-lg text-black-200 hover:bg-black-200 hover:text-white',
+    },
+    disabled: {
+      true: 'bg-gray-600 text-white cursor-default hover:bg-gray-600 hover:text-white',
+      false: '',
+    },
+  },
+  defaultVariants: {
+    variant: 'default',
+    disabled: false,
+  },
+});
+
+interface ButtonProps extends VariantProps<typeof buttonVariants> {
+  text: string;
+  onClick?: () => void;
+  className?: string;
+  disabled?: boolean;
+}
+
+export default function Button({ text, onClick, disabled = false, className, variant }: ButtonProps) {
+  const finalClassName = cn(buttonVariants({ variant, disabled }), className || '');
+
+  return (
+    <button onClick={!disabled ? onClick : undefined} disabled={disabled} className={finalClassName}>
+      {text}
```

Comment:
```suggestion
export default function Button({ onClick, disabled = false, className, variant, type }: ButtonProps) {
  const finalClassName = cn(buttonVariants({ variant, disabled }), className || '');

  return (
    <button onClick={!disabled ? onClick : undefined} disabled={disabled} className={finalClassName}>
      {children}
```
폼을 제출하게되는 페이지에서는 버튼 타입으로 submit으로도 사용되니까 type에 대한 값을 줘도 좋을 것 같아요!
그리고 버튼에 이미지가 들어갈 수도 있으니까 `{ text }` 대신 `{ children }` 으로 사용하면 어떨까요?! 사이드바에 프로필변경 버튼, 체험등록(스케쥴 추가)에 사용되는 버튼도 이미지가 들어갈거예요!